### DISCUSSION TITLE
duckdns nginx e connessione da remoto

### DISCUSSION THREAD

**Discussion by 4sp3c** [indomus/forum#2250] [HUB: Home Assistant]
ciao a tutti, ho HASSIO su un raspberry 4. Ho Installato duckdns e condigurato, ha generato le pem, ho installato e configurato nginx con https://miodominio.duckdns.org:porta... ho configurato lo configuration.yalm per indicare dove sono le pem... ho modificato il router per le porte... provo a connettermi da remoto e non riesco. come faccio a sapere se è il router che mi blocca le porta e quindi non configurato e bene oppure sono i componenti di HA? sia duckdns che nginx sul log stanno funzionando senza problemi... scusate le domande magari banali... grazie mille 

aggiungo che una volta attivato tutto, da locale potevo collegarmi con https://indirizzzo_ip_della_pi:8123 ma la prima connessione dava il messaggio di sito non sicuro

**Comment by snakuzzo** [comment]
Hai sbagliato tutto!

Se vuoi usare l'addon NGINX:
- ok a tenere l'addon DuckDNS.
- installi l'addon NGINX e ti assicuri che stia utilizzando i certificati generati tramite l'addon DuckDNS (controlla i path)
- rimuovi dal `configuration.yaml` il riferimento ai certificati

Comunque le cose non è che le devi fare a caso. Devi leggere le documentazioni di quello che vai ad installare. E'tutto scritto sulla doc dell'addon NGINX !!!!
La puoi trovare anche [QUI](https://github.com/home-assistant/addons/blob/master/nginx_proxy/DOCS.md#how-to-use)

In sostanza, la parte dei certificati la deve gestire il reverse proxy, mentre in locale devi collegarti in HTTP.

Es. se da internet voglio esporre ha sulla 443 con miodominio.duckdns.org...

L'addon lo configuri così...

![image](https://user-images.githubusercontent.com/7205371/211194546-fd170fdc-de68-47bd-9bc1-3b6ae79bd523.png)

ed il blocco `http:` nel `configuration.yaml` sarà...

```yaml
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.30.33.0/24
```

A questo punto...

Da internet -> `https://miodominio.duckdns.org`
Da LAN -> `http://ipraspberry:8123`

**Reply by 4sp3c** [reply]
> Hai sbagliato tutto!
> 
> Se vuoi usare l'addon NGINX:
> 
> * ok a tenere l'addon DuckDNS.
> * installi l'addon NGINX e ti assicuri che stia utilizzando i certificati generati tramite l'addon DuckDNS (controlla i path)
> * rimuovi dal `configuration.yaml` il riferimento ai certificati
> 
> Comunque le cose non è che le devi fare a caso. Devi leggere le documentazioni di quello che vai ad installare. E'tutto scritto sulla doc dell'addon NGINX !!!! La puoi trovare anche [QUI](https://github.com/home-assistant/addons/blob/master/nginx_proxy/DOCS.md#how-to-use)
> 
> In sostanza, la parte dei certificati la deve gestire il reverse proxy, mentre in locale devi collegarti in HTTP.
> 
> Es. se da internet voglio esporre ha sulla 443 con miodominio.duckdns.org...
> 
> L'addon lo configuri così...
> 
> ![image](https://user-images.githubusercontent.com/7205371/211194546-fd170fdc-de68-47bd-9bc1-3b6ae79bd523.png)
> 
> ed il blocco `http:` nel `configuration.yaml` sarà...
> 
> ```yaml
> http:
>   use_x_forwarded_for: true
>   trusted_proxies:
>     - 172.30.33.0/24
> ```
> 
> A questo punto...
> 
> Da internet -> `https://miodominio.duckdns.org` Da LAN -> `http://ipraspberry:8123`


ho seguito delle guide non è che faccio a caso anche perché non saprei proprio dove raccapezzarmi... e ho seguito pure video... 
comunque duck dns installato e in run e nella cartella ssl ci sono le due pem
su nginx installato e in run

![image](https://user-images.githubusercontent.com/95615624/211200150-8dc57fa6-11d8-4a43-8423-3bd713cd1414.png)

![image](https://user-images.githubusercontent.com/95615624/211200166-69cb7cff-e385-4d07-9990-c9ed08417b48.png)

su configuration.yaml avevo messo le indicazioni della posizione delle pem perché non funzionava... adesso ho messo 
![image](https://user-images.githubusercontent.com/95615624/211201560-7059a360-d7f5-4e11-a656-bfe6c51161ab.png)

continua a non farmi connettere sia che io esponga o meno le porte dal router

Il router quando espongo la 443 mi espone un'altra porta esternamente. Non so se il problema può essere quello, ma comunque se punto alla porta esposta continua a non farmi entrare. Nel caso posso bypassarla come? 

continuo di sicuro a sbagliare qualcosa ma non trovo dove... e grazie dell'aiuto

**Reply by snakuzzo** [reply]
ma cosa vorresti bypassare (termine usato a caso) ?
Se tu dici "continua a non farmi entrare" uno che cavolo dovrebbe capire ?
Ci potrebbero essere decine di cause, se non specifichi bene che errore  ti dà il browser, il server, quello che è...uno come cavolo fa ad aiutarti ? Servirebbe una sfera di cristallo!

Questo che vorrebbe dire ?

> espongo la 443 mi espone un'altra porta esternamente.

non si capisce cosa intendi.
Vuoi dire che il router non ti permette di usare la 443 ? In tal caso scegline un'altra (es.9000) e adegua:

- NGINX - mettendo la 9000 al posto della 443
- Port forwarding -> 9000 esterna -> 8123 interna

**Reply by 4sp3c** [reply]
giusto per essere chiari... bypassare termine usato giustamente che da quanto ne so significa superare o evitare un ostacolo o un'ostruzione aggirando... bypassare significa che se non la posso usare, di trovare una strada alternativa che forse me l'hai scritta alla fine che non sapevo come impostare il tutto... e il continua a non farmi entrare significa che provavo a puntare su chrome a miodominio.duckdns.org e continua a non caricarmi nulla, rimane semplicemente appeso (termine non usato di nuovo a caso). e non scrivo l'errore perché né il router, né il browser, né quello che è, va in errore ma rimane a caricare la pagina senza fare nulla di nulla
Per la porta, se attivo l'abilitazione della porta 443 (esporla) sul router, automaticamente mi scrive un'altra porta assegnata esternamente. 
provo a impostare utilizzando altre porte. Grazie comunque dell'aiuto.

**Reply by snakuzzo** [reply]
Se ti cambia la porta 443 in automatico, allora che cavolo insisti a fare con quella porta ? Non tutti i router ti permettono di farlo.
Poi scrivi...

> e il continua a non farmi entrare significa che provavo a puntare su chrome a miodominio.duckdns.org e continua a non caricarmi nulla

...grazie al cavolo (dico io).

I problemi che si hanno vanno esposti con chiarezza ed indicando tutti i dettagli.
Come hai scritto prima, per capire cosa intendevi bisognava leggerti nel cervello.

Con "continua a non farmi entrare" potevo intendere qualsiasi cosa:
- timeout (caso tuo)
- connection refused
- errore applicativo non riportato
- etc. etc.

**Reply by snakuzzo** [reply]
Scusa...questa specie di "scazzo" non è per fare i gradassi. Spesso quando voi (community) venite a chiedere aiuto qui, sul sito o sulla chat, scrivete dei topic dove chi vi risponde dovrebbe veramente leggervi la mente.
Per affrontare certe cose bisogna avere un minimo di praticità...altrimenti starvi dietro è veramente una battaglia persa.

Oltre al fatto che parlare di network, e dare supporto su reverse proxy e tutto il resto potrebbe anche essere un discorso abbastanza Off Topic

**Reply by Matty1184** [reply]
> > Hai sbagliato tutto!
> > Se vuoi usare l'addon NGINX:
> > 
> > * ok a tenere l'addon DuckDNS.
> > * installi l'addon NGINX e ti assicuri che stia utilizzando i certificati generati tramite l'addon DuckDNS (controlla i path)
> > * rimuovi dal `configuration.yaml` il riferimento ai certificati
> > 
> > Comunque le cose non è che le devi fare a caso. Devi leggere le documentazioni di quello che vai ad installare. E'tutto scritto sulla doc dell'addon NGINX !!!! La puoi trovare anche [QUI](https://github.com/home-assistant/addons/blob/master/nginx_proxy/DOCS.md#how-to-use)
> > In sostanza, la parte dei certificati la deve gestire il reverse proxy, mentre in locale devi collegarti in HTTP.
> > Es. se da internet voglio esporre ha sulla 443 con miodominio.duckdns.org...
> > L'addon lo configuri così...
> > ![image](https://user-images.githubusercontent.com/7205371/211194546-fd170fdc-de68-47bd-9bc1-3b6ae79bd523.png)
> > ed il blocco `http:` nel `configuration.yaml` sarà...
> > ```yaml
> > http:
> >   use_x_forwarded_for: true
> >   trusted_proxies:
> >     - 172.30.33.0/24
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > A questo punto...
> > Da internet -> `https://miodominio.duckdns.org` Da LAN -> `http://ipraspberry:8123`
> 
> ho seguito delle guide non è che faccio a caso anche perché non saprei proprio dove raccapezzarmi... e ho seguito pure video... comunque duck dns installato e in run e nella cartella ssl ci sono le due pem su nginx installato e in run
> 
> ![image](https://user-images.githubusercontent.com/95615624/211200150-8dc57fa6-11d8-4a43-8423-3bd713cd1414.png)
> 
> ![image](https://user-images.githubusercontent.com/95615624/211200166-69cb7cff-e385-4d07-9990-c9ed08417b48.png)
> 
> su configuration.yaml avevo messo le indicazioni della posizione delle pem perché non funzionava... adesso ho messo ![image](https://user-images.githubusercontent.com/95615624/211201560-7059a360-d7f5-4e11-a656-bfe6c51161ab.png)
> 
> continua a non farmi connettere sia che io esponga o meno le porte dal router
> 
> Il router quando espongo la 443 mi espone un'altra porta esternamente. Non so se il problema può essere quello, ma comunque se punto alla porta esposta continua a non farmi entrare. Nel caso posso bypassarla come?
> 
> continuo di sicuro a sbagliare qualcosa ma non trovo dove... e grazie dell'aiuto

Ciao,
io ho inserito nel configuration :
```
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.30.33.0/24
    - ip home assistent
    - 127.0.0.1
```

**Comment by 50494554524F** [comment]
se devi collegarti in remoto solo per homeassistant usa l'addon di tor, ora è il metodo più veloce e semplice da usare...
